nft_ingester/src/bin/dumper/main.rs

```diff
@@ -0,0 +1,232 @@
+        return Ok(());
+    };
+    let Some(last_known_fungible_key) = rocks_storage.last_known_fungible_asset_updated_key()?
+    else {
+        return Ok(());
+    };
+
+    let base_path = args
+        .dump_path
+        .unwrap_or_else(|| tempfile::TempDir::new().unwrap().path().to_path_buf());
+
+    let shards = shard_pubkeys(args.num_shards);
+    let fungible_shards = shard_pubkeys(args.fungible_num_shards);
+
+    let mut tasks = JoinSet::new();
+    let mut fungible_tasks = JoinSet::new();
+
+    for (start, end) in shards.iter() {
+        let name_postfix = if args.num_shards > 1 {
+            format!("_shard_{}_{}", start, end)
+        } else {
+            "".to_string()
+        };
+        let creators_path = base_path
+            .join(format!("creators{}.csv", name_postfix))
+            .to_str()
+            .map(str::to_owned)
+            .unwrap();
+        let assets_path = base_path
+            .join(format!("assets{}.csv", name_postfix))
+            .to_str()
+            .map(str::to_owned)
+            .unwrap();
+        let authorities_path = base_path
+            .join(format!("assets_authorities{}.csv", name_postfix))
+            .to_str()
+            .map(str::to_owned)
+            .unwrap();
+        let metadata_path = base_path
+            .join(format!("metadata{}.csv", name_postfix))
+            .to_str()
+            .map(str::to_owned)
+            .unwrap();
+        tracing::info!(
+            "Dumping to creators: {:?}, assets: {:?}, authorities: {:?}, metadata: {:?}",
+            creators_path,
+            assets_path,
+            authorities_path,
+            metadata_path,
+        );
+
+        let assets_file = File::create(assets_path.clone())
+            .map_err(|e| format!("Could not create file for assets dump: {}", e))?;
+        let creators_file = File::create(creators_path.clone())
+            .map_err(|e| format!("Could not create file for creators dump: {}", e))?;
+        let authority_file = File::create(authorities_path.clone())
+            .map_err(|e| format!("Could not create file for authority dump: {}", e))?;
+        let metadata_file = File::create(metadata_path.clone())
+            .map_err(|e| format!("Could not create file for metadata dump: {}", e))?;
+
+        let start = start.clone();
+        let end = end.clone();
+        let shutdown_rx = shutdown_rx.resubscribe();
+        let metrics = metrics.clone();
+        let rocks_storage = rocks_storage.clone();
+        tasks.spawn_blocking(move || {
+            rocks_storage.dump_nft_csv(
+                assets_file,
+                creators_file,
+                authority_file,
+                metadata_file,
+                args.buffer_capacity,
+                args.limit,
+                Some(start),
+                Some(end),
+                &shutdown_rx,
+                metrics,
+            )
+        });
+    }
+
+    for (start, end) in fungible_shards.iter() {
```

Comment:
Seems like fungible shards are waiting for nft shards to be spawned, I don't think it's necessary. Nevertheless, it will not save much time in comparison with the work that should be done.